lib/c/criu.c

```diff
@@ -1134,29 +1135,65 @@ int criu_set_page_server_address_port(const char *address, int port)
 
 static CriuResp *recv_resp(int socket_fd)
 {
+	struct msghdr msg_hdr = {0};
 	unsigned char *buf = NULL;
-	int len;
+	struct cmsghdr *cmsg;
 	CriuResp *msg = 0;
+	struct iovec io;
+	int len;
 
+	/* Check the size of the waiting data. */
 	len = recv(socket_fd, NULL, 0, MSG_TRUNC | MSG_PEEK);
 	if (len == -1) {
 		perror("Can't read request");
```

Comment:
Is it ok to print error messages from the library?

Reply:
Not sure. Another interface might have been better where the error messages are handled by the library user, but there are multiple perror()s in the library right now.